packages/medusa-plugin-contentful/src/services/contentful.js

```diff
@@ -594,124 +588,217 @@ class ContentfulService extends BaseService {
       }
     }
 
+    const ignore = await this.shouldIgnore_(variant.id, "contentful")
+    if (ignore) {
+      return Promise.resolve()
+    }
+
+    const environment = await this.getContentfulEnvironment_()
+    // check if product exists
+    let variantEntry = undefined
+    // if not, we create a new one
     try {
-      const ignore = await this.shouldIgnore_(variant.id, "contentful")
-      if (ignore) {
-        return Promise.resolve()
-      }
+      variantEntry = await environment.getEntry(variant.id)
+    } catch (error) {
+      return this.createProductVariantInContentful(variant)
+    }
 
-      const environment = await this.getContentfulEnvironment_()
-      // check if product exists
-      let variantEntry = undefined
-      // if not, we create a new one
-      try {
-        variantEntry = await environment.getEntry(variant.id)
-      } catch (error) {
-        return this.createProductVariantInContentful(variant)
-      }
+    const v = await this.productVariantService_.retrieve(variant.id, {
+      relations: ["prices", "options"],
+    })
 
-      const v = await this.productVariantService_.retrieve(variant.id, {
-        relations: ["prices", "options"],
-      })
+    const variantEntryFields = {
+      ...variantEntry.fields,
+      [this.getCustomField("title", "variant")]: {
+        "en-US": v.title,
+      },
+      [this.getCustomField("sku", "variant")]: {
+        "en-US": v.sku,
+      },
+      [this.getCustomField("options", "variant")]: {
+        "en-US": this.transformMedusaIds(v.options),
+      },
+      [this.getCustomField("prices", "variant")]: {
+        "en-US": this.transformMedusaIds(v.prices),
+      },
+      [this.getCustomField("medusaId", "variant")]: {
+        "en-US": v.id,
+      },
+    }
 
-      const variantEntryFields = {
-        ...variantEntry.fields,
-        [this.getCustomField("title", "variant")]: {
-          "en-US": v.title,
-        },
-        [this.getCustomField("sku", "variant")]: {
-          "en-US": v.sku,
-        },
-        [this.getCustomField("options", "variant")]: {
-          "en-US": this.transformMedusaIds(v.options),
-        },
-        [this.getCustomField("prices", "variant")]: {
-          "en-US": this.transformMedusaIds(v.prices),
-        },
-        [this.getCustomField("medusaId", "variant")]: {
-          "en-US": v.id,
-        },
+    variantEntry.fields = variantEntryFields
+
+    const updatedEntry = await variantEntry.update()
+    const publishedEntry = await updatedEntry.publish()
+
+    await this.addIgnore_(variant.id, "medusa")
+
+    return publishedEntry
+  }
+
+  async createProductCollectionInContentful(data) {
```

Comment:
Create Product Collection when a collection is created in Medusa

Reply:
https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/